components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/internal/IdentityCoreServiceComponent.java

```diff
@@ -147,6 +147,19 @@ protected void activate(ComponentContext ctxt) {
                 log.error("Error occurred while initializing the migration client." , e);
             }
 
+            try {
+                Class<?> migrationClientClass = Class.forName(MIGRATION_SERVICE_CLIENT_CLASS_NAME);
```

Comment:
We need to use OSGI service wiring, instead of class forName()

Reply:
+1 This is not OSGi practice , we have to use OSGi properly.

Reply:
This is changed now.